server/protocols/jmap-rfc-8621/src/test/scala/org/apache/james/jmap/http/BasicAuthenticationTokenManagerTest.scala

```diff
@@ -0,0 +1,48 @@
+package org.apache.james.jmap.http
+
+import org.apache.james.core.Domain
+import org.apache.james.dnsservice.api.DNSService
+import org.apache.james.domainlist.memory.MemoryDomainList
+import org.apache.james.jmap.http.Fixture._
+import org.apache.james.user.api.UsersRepository
+import org.apache.james.user.memory.MemoryUsersRepository
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.{BeforeEach, Test}
+import org.mockito.Mockito.mock
+
+class BasicAuthenticationTokenManagerTest {
+
+  var usersRepository: UsersRepository = _
+  var testee: BasicAuthenticationTokenManager = _
+
+  @BeforeEach
+  def setup(): Unit = {
+    val dnsService = mock(classOf[DNSService])
+    val domainList = new MemoryDomainList(dnsService)
+    domainList.addDomain(Domain.of("james.org"))
+    usersRepository = MemoryUsersRepository.withoutVirtualHosting(domainList)
+    usersRepository.addUser(username1, "password")
+    testee = new BasicAuthenticationTokenManager(usersRepository)
+  }
+
+  @Test
+  def shouldReturnAnyUsernameWhenValidBasicAuthToken(): Unit = {
+    assertThat(testee.retrieveUserNameFromCredential(userExistedToken)).isEqualTo(username1)
```

Comment:
I don't know what the val `userExistedToken` is so I cannot understand that test. I propose putting `toBase64("username:password")` instead here.

Reply:
`userExistedToken` is the user are added to MemoryUserRepository.